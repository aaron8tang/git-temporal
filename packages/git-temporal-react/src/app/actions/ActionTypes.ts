export enum ActionTypes {
  REQUEST_COMMITS = 'REQUEST_COMMITS',
  RECEIVE_COMMITS = 'RECEIVE_COMMITS',
  SELECT_PATH = 'SELECT_PATH',
  INVALIDATE_PATH = 'INVALIDATE_PATH',
  HIGHLIGHT_COMMIT = 'HIGHLIGHT_COMMIT',
  VIEW_COMMITS = 'VIEW_COMMITS',
  VIEW_FILES = 'VIEW_FILES',
  ADD_AUTHOR_FILTER = 'ADD_AUTHOR_FILTER',
  REMOVE_AUTHOR_FILTER = 'REMOVE_AUTHOR_FILTER',
  REMOVE_ALL_AUTHOR_FILTERS = 'REMOVE_ALL_AUTHOR_FILTERS',
}
