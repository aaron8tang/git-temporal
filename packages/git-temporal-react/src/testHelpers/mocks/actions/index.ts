jest.mock('app/actions', () => {
  const mockedActions = {};
  const actionsToMock = [
    'selectPath',
    'invalidatePath',
    'requestCommits',
    'receiveCommits',
    'fetchCommitsIfNeeded',
    'setAuthorsContainerSort',
    'highlightCommit',
    'viewCommits',
    'viewFiles',
  ];
  for (const methodName of actionsToMock) {
    mockedActions[methodName] = jest.fn(path => {
      return jest.fn(() => {
        return path;
      });
    });
  }
  return mockedActions;
});
