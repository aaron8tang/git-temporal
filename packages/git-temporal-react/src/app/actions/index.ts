import { ActionTypes } from './ActionTypes';

export const selectPath = path => (dispatch, _getState) => {
  // if this comes from a rename, follow the most current name
  const actualPath = path.replace(/\{(.*)\s=>\s(.*)\}/g, '$2');

  dispatch(fetchCommitsIfNeeded(actualPath));
  return { selectedPath: actualPath, type: ActionTypes.SELECT_PATH };
};

export const invalidatePath = path => ({
  selectedPath: path,
  type: ActionTypes.INVALIDATE_PATH,
});

export const addAuthorFilter = authorName => ({
  authorName,
  type: ActionTypes.ADD_AUTHOR_FILTER,
});

export const removeAuthorFilter = authorName => ({
  authorName,
  type: ActionTypes.REMOVE_AUTHOR_FILTER,
});

export const removeAllAuthorFilters = () => ({
  type: ActionTypes.REMOVE_ALL_AUTHOR_FILTERS,
});

export const highlightCommit = commitId => ({
  commitId,
  type: ActionTypes.HIGHLIGHT_COMMIT,
});

export const viewCommits = () => ({
  type: ActionTypes.VIEW_COMMITS,
});

export const viewFiles = () => ({
  type: ActionTypes.VIEW_FILES,
});

export const requestCommits = path => ({
  selectedPath: path,
  type: ActionTypes.REQUEST_COMMITS,
});

export const receiveCommits = (path, json) => ({
  selectedPath: path,
  commits: json.commits,
  type: ActionTypes.RECEIVE_COMMITS,
});

const fetchCommits = path => dispatch => {
  dispatch(requestCommits(path));
  const pathParam = path && path.trim().length > 0 ? `?path=${path}` : '';
  // TODO : replace this with serviceBaseUrl when it is in
  return fetch(`http://localhost:11966/git-temporal/history${pathParam}`)
    .then(response => response.json())
    .then(json => dispatch(receiveCommits(path, json)));
};

const shouldFetchCommits = (state, path) => {
  if (state.isFetching) {
    return false;
  }
  return (
    state.didInvalidate ||
    !state.commits ||
    state.commits.length <= 0 ||
    state.selectedPath !== path
  );
};

export const fetchCommitsIfNeeded = path => (dispatch, getState) => {
  if (shouldFetchCommits(getState(), path)) {
    return dispatch(fetchCommits(path));
  }
};
