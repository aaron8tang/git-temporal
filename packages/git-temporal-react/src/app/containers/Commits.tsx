import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  List,
  AutoSizer,
  CellMeasurer,
  CellMeasurerCache,
} from 'react-virtualized';

import { highlightCommit } from 'app/actions';
import { DispatchProps, IFilteredCommits } from 'app/interfaces';
import { getFilteredCommits } from 'app/selectors';
import { style } from 'app/styles';
import { CommitCard } from 'app/components/CommitCard';

export class Commits extends Component<IFilteredCommits & DispatchProps> {
  private _cache;
  private _list = null;

  private _listWidth = 400;
  private highlightedIndex = null;
  private scrollToIndexOnNextRender = null;
  private lastScrollToIndex = null;

  constructor(props) {
    super(props);
    this.renderRow = this.renderRow.bind(this);

    this._cache = new CellMeasurerCache({
      fixedWidth: true,
    });
  }

  componentDidUpdate(prevProps, _prevState) {
    if (this.props.commits !== prevProps.commits) {
      this.remeasureCells(prevProps.commits.length);
    }
  }

  readonly outerStyle = style('borderedPanel', {
    flexGrow: 1,
    position: 'relative',
    maxWidth: this._listWidth,
  });

  render() {
    const scrollToRow = this.scrollToIndexOnNextRender;
    const scrollToAlignment =
      this.lastScrollToIndex && this.lastScrollToIndex < scrollToRow
        ? 'center'
        : 'auto';
    return (
      <div style={this.outerStyle}>
        <AutoSizer>
          {({ height }) => {
            return (
              <List
                width={this._listWidth}
                height={height || 100}
                rowHeight={this._cache.rowHeight}
                rowRenderer={this.renderRow}
                rowCount={this.props.commits.length}
                ref={this._setListRef}
                deferredMeasurementCache={this._cache}
                scrollToIndex={scrollToRow}
                scrollToAlignment={scrollToAlignment}
              />
            );
          }}
        </AutoSizer>
      </div>
    );
  }
  renderRow({ index, key, parent, style }) {
    // console.log('render row', row);
    const commit = this.props.commits[index];
    const isHighlighted = this.props.highlightedCommitId
      ? this.props.highlightedCommitId === commit.id
      : false;
    if (isHighlighted) {
      this.highlightedIndex = index;
    }
    return (
      <CellMeasurer
        cache={this._cache}
        columnIndex={0}
        key={key}
        overscanRowCount={10}
        parent={parent}
        rowIndex={index}
        width={this._listWidth}
      >
        <div style={style}>
          <CommitCard
            commit={commit}
            index={index}
            isExpanded={isHighlighted}
            onClick={this.onCommitCardClick}
          />
        </div>
      </CellMeasurer>
    );
  }

  onCommitCardClick = (event, commit, index) => {
    event.stopPropagation();
    this.props.dispatch(highlightCommit(commit.id));
    this.remeasureCells(index);
    this.scrollToIndexOnNextRender = index;
  };

  _setListRef = ref => {
    this._list = ref;
  };

  remeasureCells(index) {
    this._cache.clear(this.highlightedIndex, 0);
    this._cache.clear(index, 0);
    this._list.recomputeRowHeights();
  }
}

export const mapStateToProps = state => {
  return getFilteredCommits(state);
};

export default connect(mapStateToProps)(Commits);
