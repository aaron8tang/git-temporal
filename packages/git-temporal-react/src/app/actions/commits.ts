import { ActionTypes } from 'app/actions/ActionTypes';
import { isVscode, vscode } from 'app/actions/vscode';
import { debug } from '@git-temporal/logger';

export const requestCommits = path => ({
  selectedPath: path,
  type: ActionTypes.REQUEST_COMMITS,
});

export const receiveCommits = (path, json) => ({
  selectedPath: path,
  commits: json.commits,
  isFileSelected: json.isFile,
  type: ActionTypes.RECEIVE_COMMITS,
});

const fetchCommits = path => dispatch => {
  dispatch(requestCommits(path));
  if (isVscode) {
    debug('sending history request to vscode ', path);
    // see actions/vscode.ts for response handling that comes as a window event
    vscode.postMessage({ path, command: 'history' });
  } else {
    const pathParam = path && path.trim().length > 0 ? `?path=${path}` : '';
    // TODO : replace this with serviceBaseUrl when it is in
    fetch(`http://localhost:11966/git-temporal/history${pathParam}`)
      .then(response => response.json())
      .then(json => dispatch(receiveCommits(path, json)));
  }
};

const shouldFetchCommits = (state, path) => {
  if (state.isFetching) {
    return false;
  }
  return (
    !state.commits || state.commits.length <= 0 || state.selectedPath !== path
  );
};

export const fetchCommitsIfNeeded = path => (dispatch, getState) => {
  if (shouldFetchCommits(getState(), path)) {
    return dispatch(fetchCommits(path));
  }
};
