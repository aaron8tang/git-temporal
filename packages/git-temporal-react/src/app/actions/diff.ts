import { ICommit } from 'app/interfaces';
import { dateFilteredCommits } from 'app/selectors/dates';
import { ActionTypes } from 'app/actions/ActionTypes';
import { isVscode, vscode } from 'app/actions/vscode';
import { debug } from '@git-temporal/logger';
import { start } from 'repl';

const requestDiff = (path: string) => ({
  selectedPath: path,
  type: ActionTypes.REQUEST_DIFF,
});

export const receiveDiff = (path: string, diff: any) => ({
  diff,
  selectedPath: path,
  type: ActionTypes.RECEIVE_DIFF,
});

export const setDiffStartCommit = (commitId: string) => ({
  commitId,
  type: ActionTypes.SET_DIFF_START_COMMIT,
});

export const setDiffEndCommit = (commitId: string) => ({
  commitId,
  type: ActionTypes.SET_DIFF_END_COMMIT,
});

export const fetchDiff = (
  path: string,
  commits: ICommit[],
  startDate: number,
  endDate: number
) => (dispatch: any): void => {
  dispatch(requestDiff(path));
  const filteredCommits = dateFilteredCommits(commits, startDate, endDate);
  if (!filteredCommits || filteredCommits.length === 0) {
    return;
  }
  // commits are in descending order
  const rightCommit = filteredCommits[0];
  const leftCommit = filteredCommits.slice(-1)[0];

  if (isVscode) {
    debug('sending diff request to vscode ', path, leftCommit, rightCommit);
    // see actions/vscode.ts for response handling that comes as a window event
    vscode.postMessage({
      path,
      leftCommit: leftCommit.id,
      rightCommit: rightCommit.id,
      command: 'diff',
    });
  } else {
    const pathParam =
      path && path.trim().length > 0 ? `?path=${path}` : '?path=.';
    const leftCommitParam = leftCommit ? `&leftCommit=${leftCommit.id}` : '';
    const rightCommitParam = rightCommit
      ? `&rightCommit=${rightCommit.id}`
      : '';

    // TODO : replace this with serviceBaseUrl when it is in
    const url = `http://localhost:11966/git-temporal/diff${pathParam}${leftCommitParam}${rightCommitParam}`;
    fetch(url)
      .then(response => response.json())
      .then(diff => dispatch(receiveDiff(path, diff)));
  }
};
