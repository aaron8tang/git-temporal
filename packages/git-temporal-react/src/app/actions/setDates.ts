import { debug } from '@git-temporal/logger';

import { setStartDate, setEndDate } from 'app/actions';
import { fetchDiff } from 'app/actions/diff';
import { ICommit } from 'app/interfaces';

export function setDates(
  dispatch: (args: any) => void,
  commits: ICommit[],
  selectedPath: string,
  startDate: number,
  endDate: number | Date | null
) {
  const epochStartDate = Math.floor(startDate / 1000);
  const epochEndDate = endDate && Math.floor((endDate as number) / 1000);

  debug('Timeplot: setDates', startDate, epochStartDate, endDate, epochEndDate);
  const [newStartDate, newEndDate] =
    epochStartDate === epochEndDate
      ? [epochStartDate, null]
      : !epochEndDate || epochStartDate < epochEndDate
        ? [epochStartDate, epochEndDate]
        : [epochEndDate, epochStartDate];

  dispatch(setStartDate(newStartDate));
  dispatch(setEndDate(newEndDate));
  dispatch(fetchDiff(selectedPath, commits, newStartDate, newEndDate));
}
