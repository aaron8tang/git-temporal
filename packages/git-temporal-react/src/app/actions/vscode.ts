import { receiveCommits, receiveCommitRange } from 'app/actions/commits';
import { receiveDiff } from 'app/actions/diff';
import { debug } from 'app/utilities/logger';

// @ts-ignore
export const isVscode = window && window.IS_VSCODE_WEBVIEW;
// @ts-ignore
export const vscode = isVscode ? acquireVsCodeApi() : null;

if (isVscode) {
  debug('running in VSCode.');
}

export function handleVscodeMessages(dispatch) {
  // @ts-ignore
  if (!window || !window.IS_VSCODE_WEBVIEW) {
    return;
  }
  window.addEventListener('message', event => {
    const { data } = event;
    debug(`actions/vscode received window message`, {
      ...data,
      // don't dump full commits or file contents to log
      commits: data.commits ? `object[${data.commits.length}]` : undefined,
      leftFileContents: redactArray(data.leftFileContents),
      rightFileContents: redactArray(data.rightFileContents),
      modifiedFiles: redactArray(data.modifiedFiles),
    });

    switch (data.type) {
      case 'commitRange':
        dispatch(receiveCommitRange(data.path, data));
        break;
      case 'history':
        dispatch(receiveCommits(data.path, data));
        break;
      case 'diff':
        dispatch(receiveDiff(data.path, data));
        break;
    }
  });
}

function redactArray(array) {
  return array ? `[..] length=${array.length}` : undefined;
}
