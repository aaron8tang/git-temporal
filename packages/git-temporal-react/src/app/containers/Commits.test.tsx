import React from 'react';
import { mount } from 'enzyme';

import { Commits, mapStateToProps } from './Commits';
import commitsForPath from 'testHelpers/mocks/commitsForPath';
import fiveCommits from 'testHelpers/mocks/fiveCommits';
import tenCommits from 'testHelpers/mocks/tenCommits';
import { CommitsContainerSorts } from 'app/actions/ActionTypes';

const filteredCommitsMock = {
  selectedPath: 'some/path',
  commits: fiveCommits,
  isFetching: false,
  didInvalidate: false,
  viewCommitsOrFiles: 'commits',
  highlightedCommitId: '56493bf1ebfab3ec102fe017f30fa4f81ba6a256',
  isFileSelected: false,
  commitsContainerSort: CommitsContainerSorts.TIME,
};

describe('containers/Commits', () => {
  describe('when rendered without props', () => {
    let wrapper;
    let mockDispatch;
    const mockEvent = { stopPropagation: jest.fn };
    beforeAll(() => {
      mockDispatch = jest.fn();
      wrapper = mount(
        <Commits dispatch={mockDispatch} {...filteredCommitsMock} />
      );
    });

    beforeEach(() => {
      mockDispatch.mockClear();
    });

    test('it should match snapshot (it should be showing empty indicator)', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('and then changing commits should render new list', () => {
      wrapper.setProps({ commits: tenCommits });
      expect(wrapper).toMatchSnapshot();
    });

    test('when clicking on a CommitCard it should call dispatch', () => {
      wrapper.instance().onCommitCardClick(mockEvent, fiveCommits[0], 0);
      expect(mockDispatch).toHaveBeenCalledTimes(1);
    });

    test('when clicking on files it should call dispatch', () => {
      wrapper.instance().onFileClick(mockEvent, 'some/bogus/file.js');
      expect(mockDispatch).toHaveBeenCalledTimes(1);
    });
  });
  describe('when calling mapStateToProps()', () => {
    test('it should respond with props', () => {
      const propsOut = mapStateToProps(commitsForPath);
      expect(propsOut).toMatchSnapshot();
    });
  });
});
