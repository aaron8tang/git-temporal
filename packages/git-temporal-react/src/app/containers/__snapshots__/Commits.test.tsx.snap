// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Commits when calling mapStateToProps() it should respond with props 1`] = `
Object {
  "commits": Array [
    Object {
      "authorDate": 1537541919,
      "authorName": "Bee Wilkerson",
      "body": "",
      "files": Array [
        Object {
          "linesAdded": 2,
          "linesDeleted": 649,
          "name": "README.md",
        },
      ],
      "hash": "c698a0f01d",
      "id": "c698a0f01de1fa682443510920a2dafc2440d6c8",
      "linesAdded": 2,
      "linesDeleted": 649,
      "message": "Explain why this fork exists in README.md",
      "relativeDate": "2 days ago",
    },
    Object {
      "authorDate": 1537272890,
      "authorName": "Tobias Nießen",
      "body": "Calling setAuthTag multiple times can result in hard to detect bugssince to the user, it is unclear which invocation actually affectedOpenSSL.PR-URL: https://github.com/nodejs/node/pull/22931Reviewed-By: Anna Henningsen <anna@addaleax.net>Reviewed-By: James M Snell <jasnell@gmail.com>Reviewed-By: Ujjwal Sharma <usharma1998@gmail.com>",
      "files": Array [
        Object {
          "linesAdded": 1,
          "linesDeleted": 1,
          "name": "doc/api/crypto.md",
        },
        Object {
          "linesAdded": 2,
          "linesDeleted": 5,
          "name": "src/node_crypto.cc",
        },
        Object {
          "linesAdded": 26,
          "linesDeleted": 0,
          "name": "test/parallel/test-crypto-authenticated.js",
        },
      ],
      "hash": "058c5b81cd",
      "id": "058c5b81cdbabe8989a194ba5d388f4c230f4af6",
      "linesAdded": 29,
      "linesDeleted": 6,
      "message": "crypto: do not allow multiple calls to setAuthTag",
      "relativeDate": "2 days ago",
    },
    Object {
      "authorDate": 1526296167,
      "authorName": "Daniel Bevenius",
      "body": "This commit adds a --completion-bash option to node which can besourced to provide bash code completion for node options.Usage:$ node --completion-bash  > node_bash_completion$ source node_bash_completion$ node --[tab]PR-URL: https://github.com/nodejs/node/pull/20713Reviewed-By: Anna Henningsen <anna@addaleax.net>Reviewed-By: James M Snell <jasnell@gmail.com>Reviewed-By: Denys Otrishko <shishugi@gmail.com>",
      "files": Array [
        Object {
          "linesAdded": 11,
          "linesDeleted": 0,
          "name": "doc/api/cli.md",
        },
        Object {
          "linesAdded": 3,
          "linesDeleted": 0,
          "name": "doc/node.1",
        },
        Object {
          "linesAdded": 25,
          "linesDeleted": 0,
          "name": "lib/internal/bash_completion.js",
        },
        Object {
          "linesAdded": 7,
          "linesDeleted": 1,
          "name": "lib/internal/bootstrap/node.js",
        },
        Object {
          "linesAdded": 1,
          "linesDeleted": 0,
          "name": "node.gyp",
        },
        Object {
          "linesAdded": 3,
          "linesDeleted": 0,
          "name": "src/node_options.cc",
        },
        Object {
          "linesAdded": 1,
          "linesDeleted": 0,
          "name": "src/node_options.h",
        },
      ],
      "hash": "56493bf1eb",
      "id": "56493bf1ebfab3ec102fe017f30fa4f81ba6a256",
      "linesAdded": 51,
      "linesDeleted": 1,
      "message": "tools: add bash completion for node",
      "relativeDate": "2 days ago",
    },
    Object {
      "authorDate": 1537212012,
      "authorName": "Michaël Zasso",
      "body": "Replace references to the outdated \`update-v8\` tool to its replacementin \`node-core-utils\`PR-URL: https://github.com/nodejs/node/pull/22913Reviewed-By: Richard Lau <riclau@uk.ibm.com>Reviewed-By: Colin Ihrig <cjihrig@gmail.com>Reviewed-By: Weijia Wang <starkwang@126.com>Reviewed-By: Sakthipriyan Vairamani <thechargingvolcano@gmail.com>Reviewed-By: Joyee Cheung <joyeec9h3@gmail.com>Reviewed-By: Luigi Pinca <luigipinca@gmail.com>Reviewed-By: James M Snell <jasnell@gmail.com>",
      "files": Array [
        Object {
          "linesAdded": 8,
          "linesDeleted": 8,
          "name": "doc/guides/maintaining-V8.md",
        },
      ],
      "hash": "408e8ce22c",
      "id": "408e8ce22ce7436e88a9977b4dc99fc3c431fc67",
      "linesAdded": 8,
      "linesDeleted": 8,
      "message": "doc: update maintaining V8 guide",
      "relativeDate": "3 days ago",
    },
    Object {
      "authorDate": 1537269982,
      "authorName": "Ruben Bridgewater",
      "body": "Currently the documentation is not specific how fast-tracking shouldbe applied. This specifies exactly how things should be done to preventconfusion.PR-URL: https://github.com/nodejs/node/pull/22929Reviewed-By: Denys Otrishko <shishugi@gmail.com>Reviewed-By: Anna Henningsen <anna@addaleax.net>Reviewed-By: Tobias Nießen <tniessen@tnie.de>Reviewed-By: Vse Mozhet Byt <vsemozhetbyt@gmail.com>Reviewed-By: John-David Dalton <john.david.dalton@gmail.com>Reviewed-By: Matteo Collina <matteo.collina@gmail.com>Reviewed-By: Luigi Pinca <luigipinca@gmail.com>Reviewed-By: Michaël Zasso <targos@protonmail.com>Reviewed-By: Weijia Wang <starkwang@126.com>Reviewed-By: Richard Lau <riclau@uk.ibm.com>Reviewed-By: Matheus Marchini <mat@mmarchini.me>Reviewed-By: Sakthipriyan Vairamani <thechargingvolcano@gmail.com>Reviewed-By: Yuta Hiroto <hello@hiroppy.me>Reviewed-By: Rich Trott <rtrott@gmail.com>Reviewed-By: Trivikram Kamat <trivikr.dev@gmail.com>Reviewed-By: George Adams <george.adams@uk.ibm.com>",
      "files": Array [
        Object {
          "linesAdded": 6,
          "linesDeleted": 3,
          "name": "COLLABORATOR_GUIDE.md",
        },
      ],
      "hash": "54af9738b9",
      "id": "54af9738b9fad1c1e46ea9170037146324cd152e",
      "linesAdded": 6,
      "linesDeleted": 3,
      "message": "doc: specify fast-tracking",
      "relativeDate": "3 days ago",
    },
  ],
  "commitsContainerSort": undefined,
  "didInvalidate": undefined,
  "highlightedCommitId": undefined,
  "isFetching": false,
  "isFileSelected": false,
  "search": "",
  "selectedPath": "testPath1",
  "viewCommitsOrFiles": "commits",
}
`;

exports[`containers/Commits when rendered without props and then changing commits should render new list 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "1px solid whitesmoke",
      "borderRadius": 3,
      "display": "flex",
      "flexDirection": "column",
      "flexGrow": 1,
      "marginRight": 10,
      "maxWidth": 400,
      "padding": 10,
      "position": "relative",
    }
  }
>
  <Connect(CommitsActionMenu) />
  <div
    style={
      Object {
        "color": "#333333",
        "display": "block",
        "flexGrow": 0,
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 20,
        "marginRight": 30,
        "position": "relative",
      }
    }
  >
    <span
      data-testId="header"
    >
      Commits by 
      time
    </span>
  </div>
  <AutoSizer
    disableHeight={false}
    disableWidth={false}
    onResize={[Function]}
    style={Object {}}
  >
    <Component />
  </AutoSizer>
</div>
`;

exports[`containers/Commits when rendered without props it should match snapshot (it should be showing empty indicator) 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "1px solid whitesmoke",
      "borderRadius": 3,
      "display": "flex",
      "flexDirection": "column",
      "flexGrow": 1,
      "marginRight": 10,
      "maxWidth": 400,
      "padding": 10,
      "position": "relative",
    }
  }
>
  <Connect(CommitsActionMenu) />
  <div
    style={
      Object {
        "color": "#333333",
        "display": "block",
        "flexGrow": 0,
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 20,
        "marginRight": 30,
        "position": "relative",
      }
    }
  >
    <span
      data-testId="header"
    >
      Commits by 
      time
    </span>
  </div>
  <AutoSizer
    disableHeight={false}
    disableWidth={false}
    onResize={[Function]}
    style={Object {}}
  >
    <Component />
  </AutoSizer>
</div>
`;
